### ISSUE TITLE
Support GHC 9.2

### ISSUE THREAD

**Issue by Bodigrim** [haskell-primitive/primitive-unlifted#28]
```
$ cabal build -w ghc-9.2
[ 1 of 18] Compiling Data.Primitive.Unlifted.Array.Primops

src/Data/Primitive/Unlifted/Array/Primops.hs:45:25: error:
    Module ‘GHC.Exts’ does not export ‘RuntimeRep(UnliftedRep)’
   |
45 |                 , TYPE, RuntimeRep (UnliftedRep), unsafeCoerce#)
```
What's worse is that when I fix imports, GHC 9.2 ends up panicking with a message beyond my comprehension.

**Comment by Bodigrim** [comment]
As a Hackage Trustee I made two revisions to prohibit unbuildable configurations:
https://hackage.haskell.org/package/primitive-unlifted-0.1.0.0/revisions/
https://hackage.haskell.org/package/primitive-unlifted-0.1.1.0/revisions/

**Comment by andrewthad** [comment]
Thanks! Also, the current approach in the 1.x series was designed by @treeowl. I am not sure if it is possible to remedy the compiler panics that it causes in newer GHCs, but he might know. The older approach in the 0.x series still works with newer versions of GHC though.

**Comment by treeowl** [comment]
Lovely. *barf*. Have you made any progress getting this legit in GHC?

**Comment by Bodigrim** [comment]
Not really, I was experimenting in `/tmp` and abandoned it once discovered that fixed imports cause GHC panic.

**Comment by wz1000** [comment]
See https://gitlab.haskell.org/ghc/ghc/-/issues/20908

**Comment by sheaf** [comment]
I don't think the uses of `unsafeCoerce#` in the implementation of unlifted `MVar`s are correct, as they coerce between `TYPE (BoxedRep Lifted)` and `TYPE (BoxedRep Unlifted)` (causing the issue Zubin linked to above).

In [GHC MR !7299](https://gitlab.haskell.org/ghc/ghc/-/merge_requests/7299) I'm adding support for levity-polymorphic `MVar#` (and arrays, `MutVar#` etc). Do you have tests or other programs that rely on unlifted `MVar#`? I want to avoid releasing a version of GHC with a subtly broken feature.

**Comment by chessai** [comment]
fixed in 9b16b79318cbe294d3fdf730b8bd0e4191bf422e

**Comment by hasufell** [comment]
Users are seeing this when compiling HLS for GHC 9.2.5.

```
Building library for primitive-unlifted-1.0.0.0..
[ 1 of 18] Compiling Data.Primitive.Unlifted.Array.Primops ( src/Data/Primitive/Unlifted/Array/Primops.hs, dist/build/Data/Primitive/Unlifted/Array/Primops.o, dist/build/Data/Primitive/Unlifted/Array/Primops.dyn_o )

src/Data/Primitive/Unlifted/Array/Primops.hs:45:25: error:
    Module ‘GHC.Exts’ does not export ‘RuntimeRep(UnliftedRep)’
   |
45 |                 , TYPE, RuntimeRep (UnliftedRep), unsafeCoerce#)
   |                         ^^^^^^^^^^^^^^^^^^^^^^^^
```
